learning-examples/get_pumpswap_pools.py

```diff
@@ -0,0 +1,86 @@
+import asyncio
+import os
+import struct
+
+import base58
+from solana.rpc.async_api import AsyncClient
+from solana.rpc.types import MemcmpOpts
+from solders.pubkey import Pubkey
+
+RPC_ENDPOINT = os.environ.get("SOLANA_NODE_RPC_ENDPOINT")
+PUMP_AMM_PROGRAM_ID = Pubkey.from_string("pAMMBay6oceH9fJKBRHGP5D4bD4sWpmSwMn52FMfXEA")
+TOKEN_MINT = Pubkey.from_string("35ySx7Rt3RqeTp75QB81FgRvPT5yDY2m5BupsUYDpump")
+
+
+async def get_market_address_by_base_mint(base_mint_address: Pubkey, amm_program_id: Pubkey):
+    async with AsyncClient(RPC_ENDPOINT) as client:
+        base_mint_bytes = bytes(base_mint_address)
+        
+        # Define the offset for base_mint field
+        offset = 43
+        
+        # Create the filter to match the base_mint
+        filters = [
+            MemcmpOpts(offset=offset, bytes=base_mint_bytes)
+        ]
+        
+        # Retrieve the accounts that match the filter
+        response = await client.get_program_accounts(
+            amm_program_id,  # AMM program ID
+            encoding="jsonParsed",
```

Comment:
let's do base64 here and it should work. this way we are following our own recommendations. btw might be good to review for json, jsonParsed, base64 and do an easy switch where possible. I'll add to the roadmap, should be an easy one but also a drastic optimization